core/ingest/src/main/java/mil/nga/giat/geowave/core/ingest/kafka/IngestFromKafkaPlugin.java

```diff
@@ -0,0 +1,50 @@
+package mil.nga.giat.geowave.core.ingest.kafka;
+
+import mil.nga.giat.geowave.core.ingest.IngestPluginBase;
+import mil.nga.giat.geowave.core.store.index.Index;
+
+/**
+ * This is the main plugin interface for ingesting intermediate data into
+ * Geowave that has previously been staged in HDFS. Although both of the
+ * available map-reduce ingestion techniques can be implemented (one that simply
+ * uses the mapper only, and another that uses the reducer as well), the
+ * framework will choose only one based on this plugin's preference, so it is
+ * unnecessary to implement both (in this case returning null is expected if its
+ * not implemented).
+ * 
+ * @param <I>
+ *            the type for intermediate data, it must match the type supported
+ *            by the Avro schema
+ * @param <O>
+ *            the type that represents each data entry being ingested
+ */
+public interface IngestFromKafkaPlugin<I, O> extends
```

Comment:
Is this used? At least the way I'm reading it AvroFormatPlugin is the one thats used?

If its not used, I suggest deleting it.